### PR TITLE
Python plugins: add default module_path to search path

### PR DESCRIPTION
**Pull request by joergsteffens**
Currently all Python modules get installed to the plugin directory.
However, every plugin configuration requires the parameter module_path set to this value.

This PR adds the Python Plugin module path to the default search path, so that specifying the module_path is only unnecessary in non-standard installations.

OP#5043

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

**Review by pstorz — COMMENTED** [review]
Checked first commit and the changes look good to me.
Good work!

File: core/src/plugins/filed/python/python-fd.cc [hunk 1]
```diff
@@ -62,10 +62,16 @@ static const int debuglevel = 150;
...
+#define PLUGIN_USAGE                                                           \
+  PLUGIN_NAME                                                                  \
+  ":module_name=<python-module-to-load>:module_path=<path-to-python-modules>:" \
+  "...\n"                                                                      \
+  "\n"                                                                         \
+  "  module_name: The name of the Python module.\n"                            \
+  "  module_path: Python search path for the module.\n"                        \
+  "               The path '" PYTHON_MODULE_PATH                               \
+  "' is always checked for modules.\n"                                         \
+  "  Additional parameter are plugin specific."
```
Comment (sduehr) [inline core/src/plugins/filed/python/python-fd.cc:74]:
```suggestion
  "  Additional parameters are plugin specific."
```

File: core/src/plugins/stored/python/python-sd.cc [hunk 2]
```diff
@@ -63,10 +63,18 @@ static const int debuglevel = 150;
...
+  ":module_name=<python-module-to-load>:module_path=<path-to-python-modules>:" \
+  "instance=<instance_id>:...\n"                                               \
+  "\n"                                                                         \
+  "  module_name: The name of the Python module.\n"                            \
+  "  module_path: Python search path for the module.\n"                        \
+  "               The path '" PYTHON_MODULE_PATH                               \
+  "' is always checked for modules.\n"                                         \
+  "  instance:    Default is ’0’.\n"                                       \
+  "               Increment the number, when using more than one plugin.\n"    \
+  "  Additional parameter are plugin specific."
```
Comment (sduehr) [inline core/src/plugins/stored/python/python-sd.cc:77]:
```suggestion
  "  Additional parameters are plugin specific."
```

File: docs/manuals/source/Appendix/Howtos.rst [hunk 3]
```diff
@@ -922,8 +922,14 @@ Configure the plugin in the |dir|:
...
          }
-         Plugin = "python:module_path=/usr/lib64/bareos/plugins:module_name=bareos-fd-mysql:db=test,wikidb"
-         #Plugin = "python:module_path=/usr/lib64/bareos/plugins:module_name=bareos-fd-mysql:mysqlhost=dbhost:mysqluser=bareos:mysqlpassword=bareos"
+         Plugin = "python"
+                  ":module_name=bareos-fd-mysql"
+                  ":db=test,wikidb"
+         #Plugin = "python"
+         #         ":module_name=bareos-fd-mysql"
+         #         ":mysqlhost=dbhost"
+         #         ":mysqluser=bareos
```
Comment (sduehr) [inline docs/manuals/source/Appendix/Howtos.rst:931]:
```suggestion
         #         ":mysqluser=bareos"
```

File: docs/manuals/source/DeveloperGuide/PythonPluginAPI.rst [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+      and was removed, so it needs to be removed in all Python callers and callbacks.
+
+   Switch to the constants being defined in the bareos[fd|sd|dir] Python plugin.
+      Import bareos[fd|sd|dir] instead of bareos_[fd|sd|dir]_consts.
+
+   Add the **st_** prefix to all members of **stat_pkt** type.
+
+   Adapt the code to run on Python 2 **and** Python 3
+      It is important to make sure the code works both for Python 2 and 3.
+      While the c++ code and the Python api have been reorganized, the Python
```
Comment (sduehr) [inline docs/manuals/source/DeveloperGuide/PythonPluginAPI.rst:266]:
```suggestion
      While the C++ code and the Python api have been reorganized, the Python
```

File: docs/manuals/source/TasksAndConcepts/Plugins/DirectorPlugins/PythonDirPlugin.rst.inc [hunk 5]
```diff
@@ -1,43 +1,40 @@
...
    single: Plugin; Python; Director
 
-The **python-dir** plugin is intended to extend the functionality of the Bareos Director by Python code. A working example is included.
+The **python-dir** (or **python3-dir**) plugin is intended
+to extend the functionality of the |dir| by Python code.
+Configuration:
 
 -  install the **bareos-director-python-plugin** package
+-  configure the :config:option:`dir/director/PluginDirectory`
+-  activate the plugin in the Bareos Director configuration (:config:option:`dir/director/PluginNames`)
```
Comment (sduehr) [inline docs/manuals/source/TasksAndConcepts/Plugins/DirectorPlugins/PythonDirPlugin.rst.inc:17]:
```suggestion
-  activate the plugin in the |dir| configuration (:config:option:`dir/director/PluginNames`)
```

File: docs/manuals/source/TasksAndConcepts/Plugins/DirectorPlugins/PythonDirPlugin.rst.inc [hunk 6]
```diff
@@ -97,5 +91,11 @@ Multiple Python Plugin Loading Example:
 Write your own Python Plugin
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Some plugin examples are available on https://github.com/bareos/bareos-contrib. The class-based approach lets you easily reuse stuff already defined in the baseclass BareosDirPluginBaseclass, which ships with the **bareos-director-python-plugin** package. The examples contain the plugin bareos-dir-nsca-sender, that submits the results and performance data of a backup job directly to Icinga:index:`\ <single: Icinga>`\  or
-Nagios:index:`\ <single: Nagios|see{Icinga}>`\  using the NSCA protocol.
+The class-based approach lets you easily reuse stuff already defined in the Python base class,
```
Comment (sduehr) [inline docs/manuals/source/TasksAndConcepts/Plugins/DirectorPlugins/PythonDirPlugin.rst.inc:94]:
```suggestion
The class-based approach lets you easily reuse code already defined in the Python base class,
```

File: docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/PythonFdPlugin.rst.inc [hunk 7]
```diff
@@ -51,7 +63,7 @@ Example:
      Include {
        Options {
          Signature = MD5 # calculate md5 checksum per file
-         Plugin = "python"
+         Plugin = "pytho3n"
```
Comment (sduehr) [inline docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/PythonFdPlugin.rst.inc:66]:
```suggestion
         Plugin = "python3"
```

**Review by sduehr — CHANGES_REQUESTED** [review]
Thanks a lot, only found some typos.

**Comment by joergsteffens** [comment]
@sduehr thank you for your suggestions. I've applied them and then rebased them on master, because of a merge conflict. It would be great, if you can review this PR again. Also 5 check-marks are not set. Do you request changes from my side?

**Comment by joergsteffens** [comment]
@sduehr one additional thing to consider is if it is appropriate to update the Python plugins version numbers from 4 to 5. Currently for all 3 Python plugins is is defined as:
```
#define PLUGIN_DATE "May 2020"
#define PLUGIN_VERSION "4"
```

**Pull request merged**